indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/pki/X509CertificateExtension.kt

```diff
@@ -3,22 +3,22 @@ package at.asitplus.signum.indispensable.pki
 import at.asitplus.signum.indispensable.asn1.*
 import at.asitplus.signum.indispensable.asn1.encoding.Asn1
 import at.asitplus.signum.indispensable.asn1.encoding.Asn1.Bool
-import kotlinx.serialization.Serializable
+import at.asitplus.signum.indispensable.pki.pkiExtensions.BasicConstraintsExtension
+import at.asitplus.signum.indispensable.pki.pkiExtensions.CertificatePoliciesExtension
+import at.asitplus.signum.indispensable.pki.pkiExtensions.InhibitAnyPolicyExtension
+import at.asitplus.signum.indispensable.pki.pkiExtensions.NameConstraintsExtension
+import at.asitplus.signum.indispensable.pki.pkiExtensions.PolicyConstraintsExtension
+import at.asitplus.signum.indispensable.pki.pkiExtensions.PolicyMappingsExtension
 
 /**
  * X.509 Certificate Extension
  */
-@ConsistentCopyVisibility
-data class X509CertificateExtension @Throws(Asn1Exception::class) private constructor(
+open class X509CertificateExtension @Throws(Asn1Exception::class) private constructor(
     override val oid: ObjectIdentifier,
-    val value: Asn1Element,
-    val critical: Boolean = false
+    open val value: Asn1Element,
+    open val critical: Boolean = false
```

Comment:
why are these `open val`s instead of the constructor being `protected` if you expect it to be invoked from subclasses